### ISSUE TITLE
ConvertIfStatementToConditionalTernaryExpressionCodeRefactoringProvider null reference exception

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#5242]
VS Mac 7.6.0.1838:

Clicking the mouse right at the end of the line below in mono-addins:

https://github.com/mono/mono-addins/blob/619d1f5f74d7f4cbd57b4f72eca2a0cb8dbc8010/Mono.Addins/Mono.Addins/ExtensionNode.cs#L414

Seems to cause the ConvertIfStatementToConditionalTernaryExpressionCodeRefactoringProvider to throw an exception and show the command bar asking to enable it.

IDE log:

```
Roslyn error: Extension_Exception ConvertIfStatementToConditionalTernaryExpressionCodeRefactoringProvider : Object reference not set to an instance of an object
  at RefactoringEssentials.TypeExtensions.GenerateTypeSyntax (Microsoft.CodeAnalysis.ITypeSymbol typeSymbol, Microsoft.CodeAnalysis.SyntaxAnnotation simplifierAnnotation) [0x00000] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/Util/TypeExtensions.cs:21 
  at RefactoringEssentials.CSharp.CodeRefactorings.ConvertIfStatementToConditionalTernaryExpressionCodeRefactoringProvider+<ComputeRefactoringsAsync>d__1.MoveNext () [0x002ed] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/ConvertIfStatementToConditionalTernaryExpressionCodeRefactoringProvider.cs:101 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
ERROR [2018-06-29 17:30:28Z]: Roslyn reported an exception to the user
System.NullReferenceException: Object reference not set to an instance of an object
  at RefactoringEssentials.TypeExtensions.GenerateTypeSyntax (Microsoft.CodeAnalysis.ITypeSymbol typeSymbol, Microsoft.CodeAnalysis.SyntaxAnnotation simplifierAnnotation) [0x00000] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/Util/TypeExtensions.cs:21 
  at RefactoringEssentials.CSharp.CodeRefactorings.ConvertIfStatementToConditionalTernaryExpressionCodeRefactoringProvider+<ComputeRefactoringsAsync>d__1.MoveNext () [0x002ed] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/ConvertIfStatementToConditionalTernaryExpressionCodeRefactoringProvider.cs:101 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
```

> VS bug [#640988](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/640988)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.1875 (release-7.6)

Pull Request #5266 merged by: Lluis Sanchez
Author: mono
Commit: 50d2e2f5ddbad8a7dd292452391a25464424a5b0 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.462 (master)

Pull Request #5265 merged by: Mike Krüger
Author: mono
Commit: b738bf3460320ce20b794df21006916ce9223009 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5265 merged by: Mike Kr??ger
Author: mono
Commit: b738bf3460320ce20b794df21006916ce9223009 (xamarin/monodevelop)
